### PR TITLE
Introduce `Set` universe WIP

### PR DESCRIPTION
**Pull request by tsao-chi**
Resolves https://github.com/aya-prover/aya-dev/issues/469

This pr:

+ renamed Univ to Sort and added different Sorts
+ replaced `lift: Int` in some places with Sort and used `Sort.max()` instead of `Math.max()`, `Sort::succ` instead of `+ 1`
+ `universe()` is renamed to `sort()` and added new api `ty()` to check sort exprs
+ implemented new tyck rules for pi and sigma
+ `I: ISet`

Todo:

- [ ] tyck rules for top level defs
- [ ] Prop tyck rules
- [ ] more Set tyck rules
- [ ] disallow data to live in `ISet` (should we?)
- [ ] #506
- [ ] more?
- [ ] irrelevance of propositions (that, if `A : Prop` and `a b : A`, then `a = b` definitionally)
- [ ] erasure of propositions (_never_ normalize a term if its type lives in `Prop`)

### CODE REVIEW

**Comment by tsao-chi** [comment]
@ice1000 no

**Comment by ice1000** [comment]
You may want to merge (not rebase!) main for the conflicts.

File: base/src/main/java/org/aya/core/term/StableWHNF.java [hunk 1]
```diff
@@ -6,15 +6,5 @@
...
-  CallTerm.Struct,
-  FormTerm.Path,
-  FormTerm.Pi,
-  FormTerm.Sigma,
-  FormTerm.Univ,
-  IntroTerm.Lambda,
-  IntroTerm.New,
-  IntroTerm.PathLam,
-  IntroTerm.Tuple {
+public sealed interface StableWHNF permits CallTerm.Data, CallTerm.Struct, FormTerm.Path, FormTerm.Pi, FormTerm.Sigma, FormTerm.Sort, IntroTerm.Lambda, IntroTerm.New, IntroTerm.PathLam, IntroTerm.Tuple {
```
Comment (ice1000) [inline base/src/main/java/org/aya/core/term/StableWHNF.java:9]: why

Comment (tsao-chi) [reply base/src/main/java/org/aya/core/term/StableWHNF.java:9]: My ide 😢

**Comment by tsao-chi** [comment]
This pr

+ renamed Univ to Sort and added different Sorts
+ replaced `lift: Int` in some places with Sort and used `Sort.max()` instead of `Math.max()`, `Sort::succ` instead of `+ 1`
+ `universe()` is renamed to `sort()` and added new api `ty()` to check sort exprs
+ implemented new tyck rules for pi and sigma
+ I: ISet

Todo:

- [ ] tyck rules for top level defs
- [ ] Prop tyck rules
- [ ] more Set tyck rules

**Review by ice1000 — APPROVED** [review]
bors r+

**Comment by ice1000** [comment]
Leftovers:

- [ ] tyck rules for top level defs
- [ ] Prop tyck rules
- [ ] more Set tyck rules
- [ ] disallow data to live in `ISet` (should we?)
- [ ] #506
- [ ] more?
- [ ] irrelevance of propositions (that, if `A : Prop` and `a b : A`, then `a = b` definitionally)
- [ ] erasure of propositions (_never_ normalize a term if its type lives in `Prop`)

**Pull request merged**